platform/commonUI/notification/src/NotificationService.js

```diff
@@ -0,0 +1,284 @@
+/*****************************************************************************
+ * Open MCT Web, Copyright (c) 2014-2015, United States Government
+ * as represented by the Administrator of the National Aeronautics and Space
+ * Administration. All rights reserved.
+ *
+ * Open MCT Web is licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * Open MCT Web includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+/*global define*/
+
+/**
+ * This bundle implements the notification service, which can be used to
+ * show banner notifications to the user. Banner notifications
+ * are used to inform users of events in a non-intrusive way. As
+ * much as possible, notifications share a model with blocking
+ * dialogs so that the same information can be provided in a dialog
+ * and then minimized to a banner notification if needed.
+ *
+ * @namespace platform/commonUI/dialog
+ */
+define(
+    ["./MessageSeverity"],
+    function (MessageSeverity) {
+        "use strict";
+
+        /**
+         * A representation of a user action. Actions are provided to
+         * dialogs and notifications and are shown as buttons.
+         *
+         * @typedef {object} NotificationAction
+         * @property {string} label the label to appear on the button for
+         * this action
+         * @property {function} action a callback function to be invoked
+         * when the button is clicked
+        */
+
+        /**
+         * A representation of a banner notification. Banner notifications
+         * are used to inform users of events in a non-intrusive way. As
+         * much as possible, notifications share a model with blocking
+         * dialogs so that the same information can be provided in a dialog
+         * and then minimized to a banner notification if needed.
+         *
+         * @typedef {object} Notification
+         * @property {string} title The title of the message
+         * @property {MessageSeverity} severity The importance of the
```

Comment:
I think that, in lieu of trying to have an enum here (which doesn't work very well in JavaScript), we could just have a string for 'info', 'alert', 'error', and mention those as options within JSDoc.

MessageSeverity.js could then either go away, or be used to map these properties to things like their associated CSS classes.

If taking this approach, from the `notify` method you could check to see if the severity level was recognized, and log a warning (then fall back to some default - maybe `error`?) if it is not.